inc/dashboard/dashboard.class.php

```diff
@@ -426,11 +456,6 @@ static function convertRights(array $raw_rights = []): array {
     * @return bool
     */
    static function checkRights(array $rights = []): bool {
-      // check global (admin) right
-      if (self::canView()) {
-         return true;
-      }
-
```

Comment:
To be checked carefully : rights checks are spread on several places.

Here I removed the canView() call because CheckRights is called in CanViewCurrent(). This method already calls canView().

In getAll() I replaced checkRights() by a call to canViewCurrent() to properly check the private status. This changes makes taht the checks includes the view all dashboards right, which was not checked before. I expect validation from reviewers.

Finally, with all the right check changes, the method checkRights may be merged with canViewCurrent() or renamed into checkShareRights().

Reply:
ping @orthagh

Reply:
should be ok